File: money/dollar.go
package money

type Dollar struct {
	amount int
}

func newDollar(amount int) *Dollar {
	d := new(Dollar)
	d.amount = amount
	return d
}

func (d *Dollar) times(multiplier int) *Dollar {
	return newDollar(d.amount * multiplier)
}

func (d *Dollar) equals(object *Dollar) bool {
	return d.amount == object.amount
}

File: python/dollar.py
from ast import Str
from typing import Any

from money import Money


class Dollar(Money):
    def __init__(self, amount: int, currency: str) -> None:
        super().__init__(amount, currency)

    def times(self, multiplier: int) -> Money:
        return Money.dollar(self.amount * multiplier)
